krux_cloud_health/cli.py

```diff
@@ -66,29 +62,30 @@ def add_cli_arguments(self, parser):
         group.add_argument(
             '--interval',
             type=str,
-            choices=INTERVAL,
-            default='daily',
+            choices=[i.name for i in Interval],
+            default=Interval.daily.name,
             help="Set time interval of data (default: %(default)s)",
         )
 
         group.add_argument(
             '--set-date',
             type=str,
             default=None,
-            help="Retrieve cost history data for specific day ('YYYY-MM-DD') or month ('YYYY-MM'), depending on interval.",
+            help="Retrieve cost history data for specific date, depending on interval. (ex: 'YYYY-MM-DD' for daily)",
         )
 
     def run(self):
         try:
-            if self.interval == 'daily':
-                cost_history = self.cloud_health.cost_history_day(self.date_input)
-            elif self.interval == 'monthly':
-                cost_history = self.cloud_health.cost_history_month(self.date_input)
-        except ValueError as e:
+            cost_history = self.cloud_health.cost_history(self.interval)
+            if self.date_input is None:
+                cost_data = cost_history.pop()
+                self.date_input = cost_data.keys()[0]
```

Comment:
Nit: I am assuming that we are setting `self.date_input` here so that we can still use the for loop at the bottom with a list of one element. Maybe calling `self.stats.incr()` here and exiting might be a better idea.